### ISSUE TITLE
Docker Daemon stopped running containers and then crashed

### ISSUE THREAD

**Issue by kostrzewa9ld** [moby/moby#37555]
**Description**

I have a Docker Swarm setup on one node. One of the service is a cron-like utility (run from `ricc/curator-cli:latest` image). Every hour it runs a container (via `docker run --rm`) to execute some short-lived command (the command should finish in a matter of seconds/minutes). At some point in time, the containers started to be created but not executed nor removed. They started to pile up until docker daemon became unresponsive and then crashed with out of memory.

**Steps to reproduce the issue:**
This issue occurred only once for me.

**Describe the results you received:**
Containers were not started. Docker daemon became unresponsive and crashed.

**Describe the results you expected:**
Containers were started.

**Additional information you deem important (e.g. issue happens only occasionally):**
I figured out that containers were created but not started from the fact that they had directories in `/var/lib/docker/containers` with `config*.json` but without a log file. Additionally I have checked that `docker run` processes had no child process reported by `ps`.

**Output of `docker version`:**

```
Client:
 Version:      17.06.2-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   cec0b72
 Built:        Tue Sep  5 19:59:06 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.2-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   cec0b72
 Built:        Tue Sep  5 20:00:25 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 38
 Running: 17
 Paused: 0
 Stopped: 21
Images: 19
Server Version: 17.06.2-ce
Storage Driver: devicemapper
 Pool Name: vg00-docker--pool
 Pool Blocksize: 524.3kB
 Base Device Size: 10.74GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 9.373GB
 Data Space Total: 21.47GB
 Data Space Available: 12.1GB
 Metadata Space Used: 2.851MB
 Metadata Space Total: 218.1MB
 Metadata Space Available: 215.3MB
 Thin Pool Minimum Free Space: 2.147GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.107-RHEL7 (2015-10-14)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: hu7zwl1siqkxv6nn2ls6zq0sp
 Is Manager: true
 ClusterID: vz8fickbr17un6c2w0i2vvfwr
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 2
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Root Rotation In Progress: false
 Node Address: xxx.xxx.xxx.xxx
 Manager Addresses:
  xxx.xxx.xxx.xxx:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 6e23458c129b551d5c9871e5174f6b1b7f6d1170
runc version: 810190ceaa507aa2727d7ae6f4790c76ec150bd2
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-327.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 62.76GiB
Name: xxx
ID: 6AYO:E25G:4CUA:BS7M:65ZM:H5E7:FPXU:QO3I:CFOD:BCG7:6Q7C:LMZ6
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
KVM VM with CentOS 7.2 (`3.10.0-327.el7.x86_64`).

[Log with oom stack trace](https://raw.githubusercontent.com/kostrzewa9ld/issues/master/moby/37555/docker.log)

**Comment by thaJeztah** [comment]
Docker CE 17.06 reached EOL in September last year; in addition, it looks like you're running an obsolete version of CentOS 7 (CentOS uses a rolling release model, so CentOS 7.2 became obsolete when 7.3 was released); are you still able to reproduce this issue on a current version of Docker and CentOS?

**Comment by kostrzewa9ld** [comment]
@thaJeztah 
My production env is currently pinned to Docker CE 17.06.2 and CentOS 7.2. Closing it for now.
